### ISSUE TITLE
Import-CrmSolution - how to increase timeout?

### ISSUE THREAD

**Issue by vk11714** [seanmcne/Microsoft.Xrm.Data.PowerShell#205]
Sorry for the silly question but I am just testing exporting/importing solutions having a hard time trying to import a solution to MSCRM online.

I am using the following command: 

**Connect-CrmOnline -ServerUrl "https://crmsandbox0000.crm.dynamics.com"
Import-CrmSolution -SolutionFilePath "C:\temp\default-fix3.zip"  -MaxWaitTimeInSeconds 6000**

Returns this error: 

**Import-CrmSolution : System.Management.Automation.RuntimeException: The request channel timed out while waiting for a reply after 00:02:00. Increase the timeout value passed to the call to Request or increase the SendTimeout value on the Binding. The time allotted to this operation may have been a portion of a longer timeout. => The HTTP request to 
 'https://crmsandbox0000.api.crm.dynamics.com/XRMServices/2011/Organization.svc' has exceeded the allotted timeout of 00:02:00. The time allotted to this operation may have been a portion of a longer timeout. => The operation has timed out**

**Comment by tjahnks23** [comment]
@vk11714 I've got the timeout to increase by declaring a param first as an Int64 then referencing it in the command. It seems like it shouldn't matter but maybe it does?? 

Param(
	[int64]$maxWaitTimeInSeconds = 1500
)
Import-CrmSolution -conn $conn -SolutionFilePath $solutionFile -MaxWaitTimeInSeconds $maxWaitTimeInSeconds

**Comment by seanmcne** [comment]
What version of the module are you currently using?  Is it 2.5?

**Comment by vk11714** [comment]
Correct

**Comment by seanmcne** [comment]
The solution import is done via async after the file upload, try it with -verbose also this isn't a default solution, is it?  How large is the zip?

**Comment by vk11714** [comment]
Hey, this is the default solution. I guess that's the issue then? I just tried with another solution and it worked fine.

**Comment by seanmcne** [comment]
Yeah it's probably the upload and parsing taking too long (though it shouldn't be impossible).  In general though I don't ever recommend importing the default solution.  You'll be far better off logically grouping things and that helps if and when you transition to managed solutions (which is far more capable than it used to be).  

Thanks for reaching out and I'm glad I could help :)

**Comment by brendandhartmannwork** [comment]
I was having this issue with a non-default solution. The problem is the Connection timeout. There is no wayto increase this with the current code base.I made a change to the psm1 file - adding a timeout parameter to **Connect-CrmOnlineDiscovery** (You may need to do this to the on-prem function if you use that) and setting it on the call to **Get-CrmConnection** via the **MaxCrmConnectionTimeOutMinutes** parameter

**Comment by seanmcne** [comment]
Thanks for the info @brendandhartmannwork!  I also have a timeout fix committed to our 2.7 codebase: https://github.com/seanmcne/Microsoft.Xrm.Data.PowerShell/pull/252  

Let me know if this helps as well, if not feel free to open a new issue.

**Comment by andigwandi** [comment]
I was having trouble with same issue, I was using Connect-CrmOnline method to connect. But I fixed the issue and am able to pass custom timeout value. I am able to import a solution to Dynamics Online which takes around 22 minutes. I hope this will help.

Here is the code:
#1 Create ConnectionString
```
$cs = "RequireNewInstance=True"
$cs+= ";Username=$($Cred.UserName)"
$cs+= ";Password=$($Cred.GetNetworkCredential().Password)"
$cs+= ";Url=$CRMServerURL"
$cs += ";AuthType=Office365"
$cs+=";SkipDiscovery=True"
```

#2 Create a connection using ConnectionString (for Online)
`$Conn = Get-CrmConnection -ConnectionString $cs -MaxCrmConnectionTimeOutMinutes $CrmConTimeOutMinutes`

#3 Import Solution
`Import-CrmSolution -SolutionFilePath $path -conn $Conn`
